### PR TITLE
Add more detail on how to build the documentation

### PR DESCRIPTION
**Pull request by KristofKoch**
**Description of proposed changes**

Added a few details on how to build the GMT documentation.

### CODE REVIEW

File: doc/rst/source/character_escape_sequences_info.rst_ [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
++----------+------------+----------+------------+
+| @s       | ß          | @i       | í          |
++----------+------------+----------+------------+
+
+Several other European accented characters and many special characters are 
+available by specifying their full octal code \\xxx (see Chapter 
+:doc:`/cookbook/octal-codes`). To use the extended character set 
+:term:`PS_CHAR_ENCODING` must be set to the respective encoding in your 
+:doc:`/gmt.conf`. Only the special characters belonging to that particular 
+encoding will be accessible. Remember: Many characters not directly available
```
Comment (joa-quim) [inline doc/rst/source/character_escape_sequences_info.rst_:58]: characters are not directly available

**Comment by KristofKoch** [comment]
I mixed up the branches, my apologies. This should be a clean pull request now. Content remained the same.

File: doc/rst/source/devdocs/contributing.rst [hunk 2]
```diff
@@ -239,16 +239,23 @@ You may need to know some basic reST syntax before making changes. Please refer
...
 
   dvc pull
   cmake --build . --target docs_depends     # Generate images included in the documentation
   cmake --build . --target optimize_images  # Optimize PNG images for documentation [optional]
-  cmake --build . --target docs_man         # UNIX manual pages
-  cmake --build . --target docs_html        # HTML manual, tutorial, cookbook, and API reference
+  cmake --build . --target docs_man         # Build UNIX manual pages
+  cmake --build . --target docs_html        # Build HTML manual, tutorial, cookbook, and API reference
+
+  cmake --build . --target install          # install the documentation into the specified location
```
Comment (maxrjones) [inline doc/rst/source/devdocs/contributing.rst:258]:
```suggestion
To install the UNIX manpages and html documentation into the specified location (along with the gmt executable, library, development headers and built-in data), use::

  cmake --build . --target install
```
I think it's helpful to specify that this command does more than just installing the documentation

Comment (KristofKoch) [reply doc/rst/source/devdocs/contributing.rst:258]: That's a good suggestion @maxrjones. Before I commit it maybe you have some deeper knowledge on the doc build & install process:

* All plots are unnecessarily generated from scratch every time (time consuming). 
Is there a way around that? I don't really need to create the plots from scratch every time when I just changed some links, layout related things or fixed a typo. Sure, `--parallel` does help but still wasted time.
* No apparent way of installing just the documentation.
It is not as time consuming as building but still not really necessary to install everything every time. 

I think that would make working on the documentation a bit more convenient.

Comment (maxrjones) [reply doc/rst/source/devdocs/contributing.rst]: These are good points @KristofKoch. unfortunately I haven't had time to look into options. Perhaps you could raise this in an issue so someone (maybe me) can look into the CMake setup later on?

Comment (seisman) [reply doc/rst/source/devdocs/contributing.rst]: > * All plots are unnecessarily generated from scratch every time (time consuming).
>   Is there a way around that? I don't really need to create the plots from scratch every time when I just changed some links, layout related things or fixed a typo. Sure, `--parallel` does help but still wasted time.

I think you don't generate the plots from scratch. The plots were saved as PS files in directories like `doc/scripts/images`. Now, these PS files are stored on the dvc remote server and can be downloaded to your computer using `dvc pull`. When building the documentation, it still needs to convert PS images to PNG images.

BTW, if you don't remove your `build` directory, running commands like `cmake --build . --target docs_html` will only rebuild the files you have changed, so it's much faster.

> * No apparent way of installing just the documentation.
>   It is not as time consuming as building but still not really necessary to install everything every time.

Could you please try if the following command works as you expect?
```
cmake --build . -DCOMPONENT=Documentation --target install
```

Comment (KristofKoch) [reply doc/rst/source/devdocs/contributing.rst]: Hi @seisman I just created a new feature request (#7225) for that as recommended by @maxrjones … just missed your comment by some minutes. Would you mind shifting the discussion over there?

Your command didn't work for me:

```
% cmake --build . --target docs_html --parallel
[…]
% cmake --build . -DCOMPONENT=Documentation --target install
Unknown argument -DCOMPONENT=Documentation
```

**Review by PaulWessel — APPROVED** [review]

**Pull request merged**